src/transformers/models/megatron_gpt2/convert_megatron_gpt2_checkpoint.py

```diff
@@ -80,6 +80,20 @@ def convert_megatron_checkpoint(args, input_state_dict, config):
     # The converted output model.
     output_state_dict = {}
 
+    nargs = input_state_dict["args"]
+    # from pprint import pprint
+    # pprint(vars(nargs))
```

Comment:
the debug prints are essential for quick troubleshooting when converting, so they are a feature and not a left-over, so I hope it's ok if they remain there. Thank you!

Reply:
Can we use the logger then and set it at a debug level?

Reply:
We don't know which parts need to be debugged ahead of time, hence everything is prepared for the developer in the various places. and a single on/off logger usually is too much information. 

The Deepspeed framework has a ton of these and they sometimes have these commented out, but most of the time they use a helper debug print util where they have a flag `force` which is `False` by default, so the code is normal but it doesn't do anything unless the developer modifies the flag to `True`.

```
print_rank0("some debug/trace", force=False)
```

Would be happy to adopt this approach at `transformers` if it resonates better. But then we need to create such a convenience function in other PR.

This is similar to:

```
        # keep for quick debug
        # print(" ".join([f"\nPYTHONPATH={self.src_dir_str}"] + cmd)); die
        execute_subprocess_async(cmd, env=self.get_env())
```

I have in various extended tests, and I use it quite often, but only when I need to. And I want the process to abort at that point. It saves a ton of time and I use it quite often when debugging these tests.

Reply:
we will get back to this subject matter and revisit the code once we agree on what the best approach is